File: algorithms/sort-algorithms/merge-sort/merge_sort.py
def merge(listA, listB):
    elements = len(listA) + len(listB)
    merged_arr = []
    for i in range(elements):
        if len(listA) == 0:
            merged_arr.append(listB[0])
            listB = listB[1:]
        elif len(listB) == 0:
            merged_arr.append(listA[0])
            listA = listA[1:]
        elif listA[0] < listB[0]:
            merged_arr.append(listA[0])
            listA = listA[1:]
        else:
            merged_arr.append(listB[0])
            listB = listB[1:]
    return merged_arr


def merge_sort(list):
    if len(list) <= 1:
        return list
    else:
        pivot_index = len(list) // 2
        left = list[:pivot_index]
        right = list[pivot_index:]
    return merge(merge_sort(left), merge_sort(right))


if __name__ == "__main__":
    pass

File: algorithms/sort-algorithms/merge-sort/mergeSort.js
function merge(arrA, arrB) {
  const elements = arrA.length + arrB.length;
  const mergedArr = [];
  for (let i = 0; i < elements; i++) {
    if (!arrA.length) {
      mergedArr.push(arrB[0]);
      arrB = arrB.slice(1, arrB.length);
    } else if (!arrB.length) {
      mergedArr.push(arrA[0]);
      arrA = arrA.slice(1, arrA.length);
    } else if (arrA[0] < arrB[0]) {
      mergedArr.push(arrA[0]);
      arrA = arrA.slice(1, arrA.length);
    } else {
      mergedArr.push(arrB[0]);
      arrB = arrB.slice(1, arrB.length);
    }
  }
  return mergedArr;
}

function mergeSort(arr) {
  if (arr.length <= 1) {
    return arr;
  } else {
    const pivot = Math.floor(arr.length / 2);
    const left = arr.slice(0, pivot);
    const right = arr.slice(pivot, arr.length);
    return merge(mergeSort(left), mergeSort(right));
  }
}
